From: Hannes Reinecke <hare@suse.de>
Subject: mapped_base for S/390
Patch-mainline: obsolete 
References: 168805

S/390 needs special care for the mapped_base patch as s390x can operate
in either 31 or 64 bit mode. When running 64 bit programs any mapped_base
is quite pointless.

Acked-by: Andrea Arcangeli <andrea@suse.de>

 processor.h |   18 ++++++++++++------
 1 files changed, 12 insertions(+), 6 deletions(-)

Index: sles10/include/asm-s390/processor.h
--- sles10/include/asm-s390/processor.h.orig	2006-05-04 15:06:51.000000000 +0200
+++ sles10/include/asm-s390/processor.h	2006-05-04 16:54:47.000000000 +0200
@@ -59,18 +59,24 @@ extern struct task_struct *last_task_use
 /*
  * User space process size: 2GB for 31 bit, 4TB for 64 bit.
  */
+#define TASK31_SIZE		(0x80000000UL)
+#define TASK64_SIZE		(0x40000000000UL)
+
 #ifndef __s390x__
 
-# define TASK_SIZE		(0x80000000UL)
-# define TASK_UNMAPPED_BASE	(TASK_SIZE / 2)
-# define DEFAULT_TASK_SIZE	(0x80000000UL)
+# define TASK_SIZE		TASK31_SIZE
+# define TASK_UNMAPPED_BASE	(current->map_base)
+# define __TASK_UNMAPPED_BASE	(TASK_SIZE / 2)
+# define DEFAULT_TASK_SIZE	TASK31_SIZE
 
 #else /* __s390x__ */
 
 # define TASK_SIZE		(test_thread_flag(TIF_31BIT) ? \
-					(0x80000000UL) : (0x40000000000UL))
-# define TASK_UNMAPPED_BASE	(TASK_SIZE / 2)
-# define DEFAULT_TASK_SIZE	(0x40000000000UL)
+					TASK31_SIZE : TASK64_SIZE)
+# define TASK_UNMAPPED_BASE	(test_thread_flag(TIF_31BIT) ? \
+					(current->map_base) : (TASK64_SIZE / 2))
+# define __TASK_UNMAPPED_BASE	(TASK31_SIZE / 2)
+# define DEFAULT_TASK_SIZE	TASK64_SIZE
 
 #endif /* __s390x__ */
 
